pkg/aws/aws.go

```diff
@@ -286,24 +286,34 @@ func getAsgs(svc *autoscaling.AutoScaling, svcEC2 *ec2.EC2, filter map[string]st
 
 	canonicalLaunchTemps := make(map[string]string)
 	if len(launchTempIds) > 0 {
-		err = svcEC2.DescribeLaunchTemplatesPages(&ec2.DescribeLaunchTemplatesInput{
-			LaunchTemplateIds: launchTempIds,
-		}, func(page *ec2.DescribeLaunchTemplatesOutput, lastPage bool) bool {
-			for _, launchTemplate := range page.LaunchTemplates {
-				if launchTemplate.LaunchTemplateId == nil {
-					return false
-				}
-				if launchTemplate.DefaultVersionNumber == nil {
-					return false
-				}
+		// DescribeLaunchTemplates limited to querying 200 at a time, so chunk the IDs
+		chunkSize := 200
+		for i := 0; i < len(launchTempIds); i += chunkSize {
+			end := i + chunkSize
+			if end > len(launchTempIds) {
+				end = len(launchTempIds)
+			}
+			chunkedLaunchTempIds := launchTempIds[i:end]
+
+			err = svcEC2.DescribeLaunchTemplatesPages(&ec2.DescribeLaunchTemplatesInput{
+				LaunchTemplateIds: chunkedLaunchTempIds,
+			}, func(page *ec2.DescribeLaunchTemplatesOutput, lastPage bool) bool {
+				for _, launchTemplate := range page.LaunchTemplates {
+					if launchTemplate.LaunchTemplateId == nil {
+						return false
+					}
+					if launchTemplate.DefaultVersionNumber == nil {
+						return false
+					}
 
-				id := *launchTemplate.LaunchTemplateId
-				canonicalLaunchTemps[id] = fmt.Sprintf("%v-%v", id, *launchTemplate.DefaultVersionNumber)
+					id := *launchTemplate.LaunchTemplateId
+					canonicalLaunchTemps[id] = fmt.Sprintf("%v-%v", id, *launchTemplate.DefaultVersionNumber)
```

Comment:
Tested against the dev cluster, there were 207 total templates, so 2 requests were made (first with 200, second with 7) and after this loop `canonicalLaunchTemps` had the full 207 templates mapped as expected